spec/services/record_declarations/base_spec.rb

```diff
@@ -132,12 +133,36 @@ def matches_lead_provider?
       )
     end
 
-    before do
-      Finance::Milestone.find_by(declaration_type: "started").update!(milestone_date: nil)
+    context "when milestone has null milestone_date" do
+      before do
+        Finance::Milestone.find_by(declaration_type: "started").update!(milestone_date: nil)
+      end
+
+      it "does not have errors on milestone_date" do
+        expect { subject.call }.not_to raise_error
+      end
     end
 
-    it "does not have errors on milestone_date" do
-      expect { subject.call }.not_to raise_error
+    context "when declaration_type does not exist for the schedule" do
+      before do
+        ect_participant_profile.schedule.milestones.find_by(declaration_type: "retained-4").destroy
+      end
+
+      subject do
+        klass.new(
+          params: {
+            course_identifier: "ecf-induction",
+            cpd_lead_provider: cpd_lead_provider,
+            declaration_date: 10.days.ago.iso8601,
+            declaration_type: "retained-4",
+            participant_id: user.id,
+          },
+        )
+      end
+
+      it "returns an error" do
+        expect { subject.call }.to raise_error(ActionController::ParameterMissing, /Declaration type does not exist for this schedule/)
+      end
     end
```

Comment:
What happens if the declaration type is missing, out of interest?

Reply:
it's covered by https://github.com/DFE-Digital/early-careers-framework/blob/develop/app/services/record_declarations/base.rb#L18 on mainline